contracts/HATVault.sol

```diff
@@ -0,0 +1,816 @@
+                revert CommitteeAlreadyCheckedIn();
+        } else {
+            if (committee != msg.sender) revert OnlyCommittee();
+        }
+
+        committee = _committee;
+
+        emit SetCommittee(_committee);
+    }
+
+    /** @notice See {IHATVault-setVestingParams}. */
+    function setVestingParams(uint256 _duration, uint256 _periods) external onlyOwner {
+        _setVestingParams(_duration, _periods);
+    }
+
+    /** @notice See {IHATVault-setBountySplit}. */
+    function setBountySplit(IHATVault.BountySplit memory _bountySplit) external onlyOwner noActiveClaim noSafetyPeriod {
+        _validateSplit(_bountySplit);
+        bountySplit = _bountySplit;
+        emit SetBountySplit(_bountySplit);
+    }
+
+    /** @notice See {IHATVault-setWithdrawalFee}. */
+    function setWithdrawalFee(uint256 _fee) external onlyFeeSetter {
+        if (_fee > MAX_FEE) revert WithdrawalFeeTooBig();
+        withdrawalFee = _fee;
+        emit SetWithdrawalFee(_fee);
+    }
+
+    /** @notice See {IHATVault-committeeCheckIn}. */
+    function committeeCheckIn() external onlyCommittee {
+        committeeCheckedIn = true;
+        emit CommitteeCheckedIn();
+    }
+
+    /** @notice See {IHATVault-setPendingMaxBounty}. */
+    function setPendingMaxBounty(uint256 _maxBounty) external onlyOwner noActiveClaim {
+        if (_maxBounty > MAX_BOUNTY_LIMIT)
+            revert MaxBountyCannotBeMoreThanMaxBountyLimit();
+        pendingMaxBounty.maxBounty = _maxBounty;
+        // solhint-disable-next-line not-rely-on-time
+        pendingMaxBounty.timestamp = block.timestamp;
+        emit SetPendingMaxBounty(_maxBounty, pendingMaxBounty.timestamp);
+    }
+
+    /** @notice See {IHATVault-setMaxBounty}. */
+    function setMaxBounty() external onlyOwner noActiveClaim {
+        if (pendingMaxBounty.timestamp == 0) revert NoPendingMaxBounty();
+
+        IHATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
+
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp - pendingMaxBounty.timestamp <
+            generalParameters.setMaxBountyDelay)
+            revert DelayPeriodForSettingMaxBountyHadNotPassed();
+        maxBounty = pendingMaxBounty.maxBounty;
+        delete pendingMaxBounty;
+        emit SetMaxBounty(maxBounty);
+    }
+
+    /** @notice See {IHATVault-setDepositPause}. */
+    function setDepositPause(bool _depositPause) external onlyOwner {
+        depositPause = _depositPause;
+        emit SetDepositPause(_depositPause);
+    }
+
+    /** @notice See {IHATVault-setVaultDescription}. */
+    function setVaultDescription(string memory _descriptionHash) external onlyOwner {
+        emit SetVaultDescription(_descriptionHash);
+    }
+
+    /** @notice See {IHATVault-setRewardController}. */
+    function setRewardController(IRewardController _newRewardController) external onlyRegistryOwner noActiveClaim {
+        rewardController.terminateVaultRewards();
```

Comment:
I want to remove this

My reasoning is this. We usually would not to change a rewardcontroller while rewards are in course. And if we do under normal circumstances, we would first stop the reward program, and then change the reward controller.

But if things go wrong, and our rewardcontroller behaves unexpected, we want to be able to change the rewardcontroller without doing a callback to the old one.

Reply:
#341